syntax = "proto3";

package rayvisiongrpc;

enum CameraType {
    HEAD = 0;
    BODY = 1;
    IR = 2;
}

enum ColorSpace
{
    RGB = 0;
    GRAY = 1;
}

message ImageData {
    int32 width = 1;
    int32 height = 2;
    ColorSpace colorspace = 3;
    bytes buffer = 4;
}

message GetImageRequest {
  CameraType type = 1;
}

message Empty {
}

// Segmentation result message
message SegmentationResult {
    repeated ImageData segments = 1;
}


service RayVisionGrpc {
  rpc GetImage(GetImageRequest) returns (ImageData);

  rpc doSegmentation(Empty) returns (stream SegmentationResult);

}